### PR TITLE
Re Org android sdk 🧱 

### PR DESCRIPTION
**Pull request by jumaallan**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

We need to deprecate use of `onSuccessJson` and `onError`, and replace this with `onResult(result: T)` which wraps both the success and error responses - and uses proper types and not pass back a string that needs to be encoded again into a type. This will make it easier for partners to implement the sdk without having to worry about any response changes from the sdk side

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
• Reorganized Android SDK code into product-specific packages
• Moved platform view classes to dedicated views package
• Consolidated mapper functions into single file
• Updated import statements across all affected files

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td><strong>SmileIDPlugin.kt</strong><dd><code>Updated imports and variable naming</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-c5073ee6b3520570f5b30e1f289ca2e7404c6c161ebbbeb733074e5002bfffd5">+36/-29</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>Mapper.kt</strong><dd><code>Added comprehensive mapper functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-cb284282a104378f40668d7d0aae4549b65bc33b2edf7475e7cf9b48bf50ba05">[link]</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDBiometricKYC.kt</strong><dd><code>Moved to biometric product package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-848ddd9ff2310201bee4c12c0461d36d6c5d8a172d1977f2eb985e8c88b27e25">+21/-20</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDDocumentCaptureView.kt</strong><dd><code>Moved to capture product package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-c55f695cf737697a16f0e2cc1485020038939495d1be689ca2c12939afad08aa">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDSmartSelfieCaptureView.kt</strong><dd><code>Moved to capture product package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-d670f1f2d4f9eb5a282f8eeac93fc3fe069ac5c89a28371408d0d4283246c2fa">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDDocumentVerification.kt</strong><dd><code>Moved to document product package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-003fe16d7ada058db13e9b36f10cb2cd5f8f8c3c9411cc885877f9f111200274">+6/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDEnhancedDocumentVerification.kt</strong><dd><code>Moved to enhanced document verification package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-9635c54c164701163a880ad3a9449000757e69f639f4d0a80d53be51bc5e0182">+10/-9</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDSmartSelfieAuthenticationEnhanced.kt</strong><dd><code>Moved to enhanced selfie package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-e50147f63e13a137d34d68ec91919b75286457b491d288e423ff4acb9afd7d04">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDSmartSelfieEnrollmentEnhanced.kt</strong><dd><code>Moved to enhanced selfie package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-354350686e3ff20a586b378e5d3210470867961804291d9bb26bb85fd1feb571">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDSmartSelfieAuthentication.kt</strong><dd><code>Moved to selfie product package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-12145d720e0a49372dfa510dcb4ace979ed4b78fd9ccb4731fdea1b5f5f94ae8">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDSmartSelfieEnrollment.kt</strong><dd><code>Moved to selfie product package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-5ed50323fce27c876b1c86060138bcdc4b606a1f255498004cd8a28443796df0">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileComposablePlatformView.kt</strong><dd><code>Moved to views package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-0bac23b7f144b742a36f70b7ba8208e8bf3f761790a20febeedd7cdd17f9263d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileSelfieComposablePlatformView.kt</strong><dd><code>Moved to views package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/flutter/pull/180/files#diff-9b04eb7a9cea351ca23f8e71bb034a7be60da4313b62f0dfdb426c3681b1b89d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: android/src/main/kotlin/com/smileidentity/flutter/SmileIDPlugin.kt [hunk 1]
```diff
@@ -241,17 +248,17 @@ class SmileIDPlugin :
...
-                        )
-                    },
+                        ),
+                    livenessImages =
+                        livenessImages.map {
+                            File(selfieImage).asFormDataPart(
+                                partName = "liveness_images",
+                                mediaType = "image/jpeg",
+                            )
+                        },
```
[the remark this answers is not in the document]
Comment (jumaallan) [reply android/src/main/kotlin/com/smileidentity/flutter/SmileIDPlugin.kt]: fixed

**Review by wangerekaharun — APPROVED** [review]
